src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +781,262 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
```

Comment:
Question: Why do we need to introduce the `script_args` item that contains an array of `in_footer` and `strategy`? Wouldn't it be simpler to just keep `group` as-is (being exposed as `in_footer` in the `$args` param of `wp_enqueue_script()`/`wp_register_script()`). Otherwise, there could be a scenario where the `data.script_args.in_footer` gets out of sync with the `data.group`, leading to confusion due to the duplicated data.

Then the `strategy` can be added as a sibling top-level key.

This entire method override could be removed, and in the `get_intended_strategy` method  we can replace this logic:

```php
$script_args = $this->get_data( $handle, 'script_args' );
$strategy    = isset( $script_args['strategy'] ) ? $script_args['strategy'] : '';
```

With simply:

```php
$strategy = $this->get_data( $handle, 'strategy' );
```

Reply:
PR with suggestion applied: https://github.com/10up/wordpress-develop/pull/64

Reply:
Thanks @westonruter the referenced PR (https://github.com/10up/wordpress-develop/pull/64) has been approved on my end, I'm really happy with this change :)